sdk/search/Azure.Search/src/Utilities/SearchExtensions.cs

```diff
@@ -47,5 +48,25 @@ public static string GetSearchServiceName(this Uri endpoint)
                 host.Substring(0, separator) :
                 null;
         }
+
+        /// <summary>
+        /// Join a collection of strings into a single comma separated string.
+        /// If the collection is null or empty, a null string will be returned.
+        /// </summary>
+        /// <param name="items">The items to join.</param>
+        /// <returns>The items joined together by commas.</returns>
+        public static string CommaSeparate(this ICollection<string> items) =>
+            items?.Count > 0 ? string.Join(",", items) : null;
+
+        /// <summary>
+        /// Split a collection of strings by commas.
+        /// </summary>
+        /// <param name="value">The value to split.</param>
+        /// <returns>A collection of individual values.</returns>
+        public static IList<string> CommaSplit(string value) =>
+            string.IsNullOrEmpty(value) ?
+                new List<string>() :
```

Comment:
Does this need to possibly be mutable? In other projects, I keep a type-cached array (it's what `Enumerable.Empty` does internally, but exposes as an `IEnumerable<T>` - I think `Array` has something though) for this purpose if immutable. Less memory overhead.

Reply:
Yes - we can give it back to users, they can add some values, then send it back again.